src/Orleans/AzureUtils/OrleansSiloInstanceManager.cs

```diff
@@ -168,15 +168,18 @@ private OrleansSiloInstanceManager(string deploymentId, string storageConnection
             {
                 await instance.storage.InitTableAsync()
                     .WithTimeout(initTimeout);
-
             }
-            catch (TimeoutException)
+            catch (TimeoutException te)
             {
-                instance.logger.Fail(ErrorCode.AzureTable_32, String.Format("Unable to create or connect to the Azure table in {0}", initTimeout));
+                string errorMsg = String.Format("Unable to create or connect to the Azure table in {0}", initTimeout);
+                instance.logger.Error(ErrorCode.AzureTable_32, errorMsg, te);
+                throw new OrleansException(errorMsg, te);
```

Comment:
It's not clear to me if it's safe to assume that switching from terminating the silo process to throwing an exception will have no impact on the run time case. Maybe it is, but looking at the code it's not obvious to me.